### PR TITLE
Add dotnet formatters info to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by donnie-msft**
Dotnet formatters were enabled recently, but I believe we need to indicate this to the community.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,6 +85,15 @@ NuGet members may contribute directly to the main remote.
 
     > Note: You have to to run .\configure.ps1 and .\build.ps1 at least once in order for your build to succeed.
 
+1. Run dotnet code formatters and correct any errors.
```
Comment (nkolev92) [inline CONTRIBUTING.md:94]: Not sure this needs to be an extra step, before writing the tests. 

If anything it'd be after. 

I'd recommending adding the check to the build.ps1 script so people locally can repro what we do on the CI. 

This is *not* the only issue people might run into.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -85,6 +85,15 @@ NuGet members may contribute directly to the main remote.
...
+    * You can use `Format Document` in VS: 
+
+       `Ctrl+K, Ctrl+D` or Edit > Advanced > Format Document (https://docs.microsoft.com/visualstudio/ide/default-keyboard-shortcuts-in-visual-studio?view=vs-2019#text-editor)
+
+    * You can use the dotnet CLI tool (https://github.com/dotnet/format):
+
+      `dotnet format --check --exclude submodules`
```
Comment (nkolev92) [inline CONTRIBUTING.md]: `packages` and `cli` need to be excluded as well.

File: CONTRIBUTING.md (hunk 1 above)
Comment (donnie-msft) [reply CONTRIBUTING.md:94]: You mean before running the tests?
I would argue you don't want to run tests, then have to modify code due to formatting. Why wait?

File: CONTRIBUTING.md (hunk 2 above)
Comment (donnie-msft) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 1 above)
Comment (loic-sharma) [reply CONTRIBUTING.md:94]: Ideally contributors should format as a final step when *all* code changes are completed. Running tests may uncover test failures that you'll have to fix, after which you'll have to format again.

Comment (kartheekp-ms) [reply CONTRIBUTING.md:94]: We can consider adding this as an opt-in feature in the build script if we don't want to break local builds due to formatting every time. By default, we run format check and community members can invoke something like `build.ps1 -skipFormatCheck` to skip format check during inner loop.

Comment (donnie-msft) [reply CONTRIBUTING.md:94]: I'm not sure I want to run all my tests, only to find out I need to change a bunch of code files for formatting purposes and have to rerun them.
But I don't care where we put it as long as it's at least mentioned. Devs will do their process however they want...

Comment (donnie-msft) [reply CONTRIBUTING.md:94]: Moved to last step

File: CONTRIBUTING.md (hunk 2 above)
Comment (zivkan) [reply CONTRIBUTING.md]: I don't believe it does need to exclude cli and packages. When running `dotnet format`, it clearly says it's checking the projects in `nuget.sln`, so it's not searching for all code files, it's searching for code files used by projects used by nuget.sln (meaning that the Apex projects and source code are not being checked).

**Pull request merged**